master/buildbot/status/buildstep.py

```diff
@@ -267,6 +271,10 @@ def addURL(self, name, url, results=None):
         if results is not None:
             self.urls[name] = {'url': url, 'results': results}
 
+    def addArtifactURL(self, name, url, results=None):
+        self.artifactUrls[name] = url
+        if results is not None:
+            self.artifactUrls[name] = {'url': url, 'results': results}
```

Comment:
Investigating more, it seems that - since the results value is only set by dependencies - it'd be better to split them into a third category (DependencyLinks or DependencyUrls or such) now while we're refactoring the URLs/Artifacts.